projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts

```diff
@@ -307,6 +343,14 @@ export class RasterSymbologyEditorComponent implements OnChanges, OnDestroy, Aft
         this.updateLayerMaxValue(breakpoints[breakpoints.length - 1].value);
     }
 
```

Comment:
![grafik](https://user-images.githubusercontent.com/4947061/205032065-a6154d00-8837-4c16-a6e4-83587d5cd2aa.png)

The empty space and the button somehow looks bad or broken, because it looks like we just randomly put a button there.

Can we change that somehow that it looks good with and without a histogram plot being present?

Maybe it is sufficient to align the button to the right or left or reduce the empty space ontop… I'm not sure. But we shouldn't leave it as it because the rest looks good now.

Reply:
Changed it so at first there is no empty space, the button is left aligned and says "Create Histogram". When the first histogram is created, the button is then right aligned below the histogram and says "Update Histogram" as before
<img width="434" alt="image" src="https://user-images.githubusercontent.com/57327151/205598400-e7d46843-13a0-44c9-b4cf-d798ee4d9a30.png">

[c618e4d](https://github.com/geo-engine/geoengine-ui/pull/513/commits/c618e4d5884fe8dc8d72f3f266b4212c84d8c86e)

Reply:
diff --git a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
index 5f5cbd45..fb7f7e97 100644
--- a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
+++ b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
@@ -111,17 +111,16 @@
                         </ng-container>
                         <mat-progress-bar mode="indeterminate" *ngIf="histogramLoading | async"></mat-progress-bar>
                     </div>
-                    <div class="actions" *ngIf="histogramCreated; then createdBlock; else notCreatedBlock"></div>
-                    <!-- Only displayed before any histogram is created -->
-                    <ng-template #createdBlock>
-                        <div class="actions">
-                            <button mat-raised-button class="button" color="primary" (click)="updateHistogram()">Update Histogram</button>
-                        </div>
-                    </ng-template>
-                    <!-- Only displayed after histogram is created -->
-                    <ng-template #notCreatedBlock>
-                        <button mat-raised-button class="button" color="primary" (click)="updateHistogram()">Create Histogram</button>
-                    </ng-template>
+                    <div class="histogram-actions">
+                        <button mat-stroked-button class="button" (click)="updateHistogram()">
+                            <ng-container *ngIf="histogramCreated; then createdBlock; else notCreatedBlock"></ng-container>
+                            <!-- Only displayed before any histogram is created -->
+                            <ng-template #createdBlock>Update Histogram</ng-template>
+                            <!-- Only displayed after histogram is created -->
+                            <ng-template #notCreatedBlock>Create Histogram</ng-template>
+                        </button>
+                    </div>
+
                     <mat-divider></mat-divider>
 
                     <geoengine-color-map-selector
@@ -149,9 +148,7 @@
     </div>
 
     <div class="actions">
-        <button mat-raised-button color="accent" class="button" (click)="resetChanges(layer)" [disabled]="unappliedChanges === false">
-            Reset
-        </button>
+        <button mat-raised-button class="button" (click)="resetChanges(layer)" [disabled]="unappliedChanges === false">Reset</button>
 
         <button mat-raised-button color="primary" class="button" (click)="applyChanges()" [disabled]="unappliedChanges === false">
             Apply
diff --git a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
index dd88bd7a..156b39bf 100644
--- a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
+++ b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
@@ -20,6 +20,10 @@ mat-slider {
     }
 }
 
+.histogram-actions {
+    text-align: center;
+}
+
 #gradientType {
     flex-wrap: wrap;
 }

Reply:
This still looks like the button was placed by accident.

What about this?

```diff
diff --git a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
index 5f5cbd45..fb7f7e97 100644
--- a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
+++ b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html
@@ -111,17 +111,16 @@
                         </ng-container>
                         <mat-progress-bar mode="indeterminate" *ngIf="histogramLoading | async"></mat-progress-bar>
                     </div>
-                    <div class="actions" *ngIf="histogramCreated; then createdBlock; else notCreatedBlock"></div>
-                    <!-- Only displayed before any histogram is created -->
-                    <ng-template #createdBlock>
-                        <div class="actions">
-                            <button mat-raised-button class="button" color="primary" (click)="updateHistogram()">Update Histogram</button>
-                        </div>
-                    </ng-template>
-                    <!-- Only displayed after histogram is created -->
-                    <ng-template #notCreatedBlock>
-                        <button mat-raised-button class="button" color="primary" (click)="updateHistogram()">Create Histogram</button>
-                    </ng-template>
+                    <div class="histogram-actions">
+                        <button mat-stroked-button class="button" (click)="updateHistogram()">
+                            <ng-container *ngIf="histogramCreated; then createdBlock; else notCreatedBlock"></ng-container>
+                            <!-- Only displayed before any histogram is created -->
+                            <ng-template #createdBlock>Update Histogram</ng-template>
+                            <!-- Only displayed after histogram is created -->
+                            <ng-template #notCreatedBlock>Create Histogram</ng-template>
+                        </button>
+                    </div>
+
                     <mat-divider></mat-divider>
 
                     <geoengine-color-map-selector
@@ -149,9 +148,7 @@
     </div>
 
     <div class="actions">
-        <button mat-raised-button color="accent" class="button" (click)="resetChanges(layer)" [disabled]="unappliedChanges === false">
-            Reset
-        </button>
+        <button mat-raised-button class="button" (click)="resetChanges(layer)" [disabled]="unappliedChanges === false">Reset</button>
 
         <button mat-raised-button color="primary" class="button" (click)="applyChanges()" [disabled]="unappliedChanges === false">
             Apply
diff --git a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
index dd88bd7a..156b39bf 100644
--- a/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
+++ b/projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.scss
@@ -20,6 +20,10 @@ mat-slider {
     }
 }
 
+.histogram-actions {
+    text-align: center;
+}
+
 #gradientType {
     flex-wrap: wrap;
 }

```

Cf. `git apply` for applying this.

Reply:
Ok, I applied these changes in [31041bf](https://github.com/geo-engine/geoengine-ui/pull/513/commits/31041bf8a9dd6d3f27a9956618b1e6fb56c9cebb)